src/main/java/com/k3project/demo/controller/AuthController.java

```diff
@@ -0,0 +1,77 @@
+package com.k3project.demo.controller;
+
+import com.k3project.demo.entity.Role;
+import com.k3project.demo.entity.UserEntity;
+import com.k3project.demo.repository.RoleRepository;
+import com.k3project.demo.repository.UserRepository;
+import com.k3project.demo.security.JWTGenerator;
+import com.k3project.demo.service.dto.AuthResponseDTO;
+import com.k3project.demo.service.dto.LoginDTO;
+import com.k3project.demo.service.dto.RegisterDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Collections;
+
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
```

Comment:
Add logger @Slf4j and use log in the method.

Reply:
talk